### PR TITLE
doc/add_api_comments

### PR DESCRIPTION
**Pull request by alhiwatan**
This PR includes:

- Added comments to the public APIs and fields.
- Exported ad_classes.dart for dartdoc to generate.

Tried to follow the dart doc guide:

https://dart.dev/guides/language/effective-dart/documentation

### CODE REVIEW

File: applovin_max/lib/applovin_max.dart [hunk 1]
```diff
@@ -107,116 +120,172 @@ class AppLovinMAX {
...
   }
 
-  ///
-  /// BANNERS
-  ///
+  //
+  // BANNERS
+  //
 
+  /// Sets an [AdViewAdListener] listener that notifies ad events.
```
Comment (PLP-Consulting) [inline applovin_max/lib/applovin_max.dart:245]: How about:
> Sets an [AdViewAdListener] listener with which you can receive notifications about ad events.

File: applovin_max/lib/applovin_max.dart [hunk 2]
```diff
@@ -225,91 +294,117 @@ class AppLovinMAX {
...
+  /// [](https://dash.applovin.com/documentation/mediation/flutter/getting-started/banners#displaying-a-banner)
   static void showBanner(String adUnitId) {
     channel.invokeMethod('showBanner', {
       'ad_unit_id': adUnitId,
     });
   }
 
+  /// Hides the banner for a given ad unit id.
+  ///
+  /// [](https://dash.applovin.com/documentation/mediation/flutter/getting-started/banners#displaying-a-banner)
```
Comment (PLP-Consulting) [inline applovin_max/lib/applovin_max.dart:308]: Missing link text

File: applovin_max/lib/src/ad_listeners.dart [hunk 3]
```diff
@@ -1,24 +1,30 @@
...
     required this.onAdLoadedCallback,
     required this.onAdLoadFailedCallback,
     required this.onAdClickedCallback,
   });
 }
 
-// Fullscreen Ad Listener
+/// Defines a fullscreen ad listener.
 abstract class FullscreenAdListener extends AdListener {
   final Function(MaxAd ad) onAdDisplayedCallback;
```
Comment (PLP-Consulting) [inline applovin_max/lib/src/ad_listeners.dart:23]: These three functions are missing documentation

**Review by PLP-Consulting — CHANGES_REQUESTED** [review]

File: applovin_max/lib/applovin_max.dart [hunk 4]
```diff
@@ -107,116 +120,172 @@ class AppLovinMAX {
...
   }
 
+  //
+  // PRIVACY APIs
+  //
+
+  /// Returns a consent dialog state for the current user.
   ///
-  /// PRIVACY APIs
+  /// See [ConsentDialogState] for the state of a consent dialog.
```
[the remark this answers is not in the document]
Comment (alhiwatan) [reply applovin_max/lib/applovin_max.dart:146]: Would this work?

```
  /// Returns an integer that encodes the state of the consent dialog.
  ///
  /// To learn more about how this information is encoded in the integer, see [ConsentDialogState].
  ///
  /// [Consent Flags in GDPR and Other Regions](https://dash.applovin.com/documentation/mediation/flutter/getting-started/privacy#consent-flags-in-gdpr-and-other-regions)
  static Future<int?> getConsentDialogState() {
    return channel.invokeMethod('getConsentDialogState');
  }
```

File: applovin_max/lib/applovin_max.dart [hunk 5]
```diff
@@ -107,116 +120,172 @@ class AppLovinMAX {
...
   ///
-
+  /// [Consent Flags in GDPR and Other Regions](https://dash.applovin.com/documentation/mediation/flutter/getting-started/privacy#consent-flags-in-gdpr-and-other-regions)
   static Future<int?> getConsentDialogState() {
     return channel.invokeMethod('getConsentDialogState');
   }
 
+  /// Sets whether or no the user has provided consent for interest-based advertising.
```
[the remark this answers is not in the document]
Comment (alhiwatan) [reply applovin_max/lib/applovin_max.dart:153]: done

File: applovin_max/lib/applovin_max.dart [hunk 6]
```diff
@@ -107,116 +120,172 @@ class AppLovinMAX {
...
   }
 
+  /// Checks if the user is age-restricted.
+  ///
+  /// [Prohibition on Personal Information from Children](https://dash.applovin.com/documentation/mediation/flutter/getting-started/privacy#prohibition-on-personal-information-from-children)
   static Future<bool?> isAgeRestrictedUser() {
     return channel.invokeMethod('isAgeRestrictedUser');
   }
 
+  /// Sets whether or not the user has opted out of interest-based advertising.
```
[the remark this answers is not in the document]
Comment (alhiwatan) [reply applovin_max/lib/applovin_max.dart:185]: Updated.

File: applovin_max/lib/applovin_max.dart [hunk 7]
```diff
@@ -107,116 +120,172 @@ class AppLovinMAX {
...
+  ///
+  /// [Setting an Ad Placement Name](https://dash.applovin.com/documentation/mediation/features/s2s-impression-revenue-api#setting-an-ad-placement-name)
   static void setBannerPlacement(String adUnitId, String placement) {
     channel.invokeMethod('setBannerPlacement', {
       'ad_unit_id': adUnitId,
       'placement': placement,
     });
   }
 
+  /// Updates the banner position for a given ad unit id.
```
[the remark this answers is not in the document]
Comment (thomasmso) [reply applovin_max/lib/applovin_max.dart:280]: cc @christophercong re: recent api doc changes with Ad Unit ID references

File: applovin_max/lib/applovin_max.dart [hunk 8]
```diff
@@ -225,91 +294,117 @@ class AppLovinMAX {
...
+  /// Hides the banner for a given ad unit id.
+  ///
+  /// [](https://dash.applovin.com/documentation/mediation/flutter/getting-started/banners#displaying-a-banner)
   static void hideBanner(String adUnitId) {
     channel.invokeMethod('hideBanner', {
       'ad_unit_id': adUnitId,
     });
   }
 
+  /// Destroys the banner for a given ad unit id.
```
[the remark this answers is not in the document]
Comment (thomasmso) [reply applovin_max/lib/applovin_max.dart:315]: (cleans up system resources allocated for the banner)

File: applovin_max/lib/applovin_max.dart [hunk 9]
```diff
@@ -107,116 +120,172 @@ class AppLovinMAX {
...
   }
 
-  ///
-  /// GENERAL PUBLIC API
-  ///
+  //
+  // GENERAL PUBLIC API
+  //
 
+  /// Sets your own internal user ID to the current user for the MAX S2S Rewarded Callback requests.
```
[the remark this answers is not in the document]
Comment (alhiwatan) [reply applovin_max/lib/applovin_max.dart:205]: Updated.

File: applovin_max/lib/applovin_max.dart [hunk 10]
```diff
@@ -107,116 +120,172 @@ class AppLovinMAX {
...
+  ///
+  /// [California Consumer Privacy Act (“CCPA”)](https://dash.applovin.com/documentation/mediation/flutter/getting-started/privacy#california-consumer-privacy-act-(%E2%80%9Cccpa%E2%80%9D))
   static void setDoNotSell(bool isDoNotSell) {
     channel.invokeMethod('setDoNotSell', {
       'value': isDoNotSell,
     });
   }
 
+  /// Check if the user has opted out of interest-based advertising.
+  ///
```
[the remark this answers is not in the document]
Comment (alhiwatan) [reply applovin_max/lib/applovin_max.dart:195]: Updated.

File: applovin_max/lib/applovin_max.dart (hunk 1 above)
Comment (alhiwatan) [reply applovin_max/lib/applovin_max.dart:245]: Updated.

File: applovin_max/lib/applovin_max.dart [hunk 11]
```diff
@@ -107,116 +120,172 @@ class AppLovinMAX {
...
 
+  /// Updates the banner position for a given ad unit id.
   static void updateBannerPosition(String adUnitId, AdViewPosition position) {
     channel.invokeMethod('updateBannerPosition', {
       'ad_unit_id': adUnitId,
       'position': position.value,
     });
   }
 
+  /// Sets an extran parameter to the banner for a given ad unit id.
```
[the remark this answers is not in the document]
Comment (alhiwatan) [reply applovin_max/lib/applovin_max.dart:288]:
```
  /// Sets an extra parameter to the banner for a given ad unit identification.
  ///
  /// For example, pass "adaptive_banner" and "false" to this method as the key/value pair
  /// to disable Adaptive Banners for the specified [adUnitId.
  ///
  /// [Adaptive Banners](https://dash.applovin.com/documentation/mediation/flutter/getting-started/banners#adaptive-banners)
  static void setBannerExtraParameter(String adUnitId, String key, String value) {
    channel.invokeMethod('setBannerExtraParameter', {
      'ad_unit_id': adUnitId,
      'key': key,
      'value': value,
    });
  }
```

**Comment by alhiwatan** [comment]
Have updated the comments, so please review them.  And I have one question.

We have two styles of specifying ad unit id,  'with the specified [adUnitId]' and 'for a given ad unit identification'.   For example.

```
1.  /// Sets an ad placement name for the MREC with the specified [adUnitId].
2.  /// Updates the MREC position for a given ad unit identification.
```

Either way looks okay but which one should we go for?   Or does it depend?

**Comment by PLP-Consulting** [comment]
> Have updated the comments, so please review them. And I have one question.
> 
> We have two styles of specifying ad unit id, 'with the specified [adUnitId]' and 'for a given ad unit identification'. For example.
> 
> ```
> 1.  /// Sets an ad placement name for the MREC with the specified [adUnitId].
> 2.  /// Updates the MREC position for a given ad unit identification.
> ```
> 
> Either way looks okay but which one should we go for? Or does it depend?

I'd recommend using the same phrasing across all the various methods. Of those two, I prefer the first one ("...with the specified [adUnitId].")

File: applovin_max/lib/src/ad_classes.dart [hunk 12]
```diff
@@ -1,12 +1,19 @@
+/// Represents an ad that has been served by AppLovin MAX.
 class MaxAd {
+  /// The ad unit ID for which this ad was loaded.
   final String adUnitId;
+  /// The ad network from which this ad was loaded.
   final String networkName;
+  /// The ad’s revenue amount, or 0 if no revenue amount exists.
   final String revenue;
+  /// The creative id tied to the ad, if any. You can report creative issues to the corresponding ad network using this id.
```
Comment (PLP-Consulting) [inline applovin_max/lib/src/ad_classes.dart:9]: Looks like there are still a few lowercased "id"s.

**Comment by alhiwatan** [comment]
Updated, please review.

**Review by PLP-Consulting — APPROVED** [review]

**Review by thomasmso — APPROVED** [review]
Looks amazing, thanks @PLP-Consulting / @alhiwatan

**Pull request merged**